runtime/gc_include/ObjectAccessBarrierAPI.hpp

```diff
@@ -2989,6 +3011,13 @@ class MM_ObjectAccessBarrierAPI
 	}
 #endif /* J9VM_GC_GENERATIONAL */
 
+	static VMINLINE bool
+	usingSATB(J9VMThread *vmThread)
+	{
+		UDATA writeBarrierType = vmThread->javaVM->gcWriteBarrierType;
```

Comment:
I can see the challenge....
Another way is to add an instance of MM_ObjectAccessBarrierAPI as a non-static  member of MM_ObjectAllocationAPI, and make all these new helpers non-static, what's more in line how these APIs are meant to be used.